src/featuremap/ugbio_featuremap/somatic_featuremap_fields_transformation.py

```diff
@@ -0,0 +1,578 @@
+
+
+def collapse_bed_by_chunks(bed_file: str, num_chunks: int) -> list[str]:
+    """
+    Collapse a sorted BED file into equal-sized chunks by number of rows.
+
+    This function reads a BED file and divides it into approximately equal-sized chunks
+    based on the number of rows. Each chunk is represented by a single genomic interval
+    spanning from the first to the last position within that chunk.
+
+    Parameters
+    ----------
+    bed_file : str
+        Path to input sorted BED file containing columns (chrom, start, end).
+    num_chunks : int
+        Number of chunks to create. If the number of rows in the BED file
+        is less than num_chunks, the actual number of chunks will be
+        adjusted accordingly.
+
+    Returns
+    -------
+    list of str
+        List of genomic intervals in the format "chrom:start-end",
+        where each interval represents a collapsed chunk.
+
+    Notes
+    -----
+    - The function assumes the input BED file has no header and contains
+      exactly 3 columns: chromosome, start position, and end position.
+    - Chunks are created based on equal distribution of rows, not equal
+      genomic distance.
+    - Each chunk preserves the chromosome of its first row; this assumes
+      chunks don't span multiple chromosomes.
+
+    Examples
+    --------
+    >>> intervals = collapse_bed_by_chunks("input.bed", num_chunks=4)
+    >>> print(intervals)
+    ['chr1:1000-5000', 'chr1:5001-10000', 'chr2:1000-8000', 'chr2:8001-15000']
+    """
+    df_bed_regions = pd.read_csv(bed_file, sep="\t", header=None, names=["chrom", "start", "end"], usecols=[0, 1, 2])
+    n = len(df_bed_regions)
+    # Handle case where rows < chunks: adjust num_chunks
+    num_chunks = min(num_chunks, n)
+    # Compute chunk size (number of rows per chunk, roughly equal)
+    chunk_size = math.ceil(n / num_chunks)
+    # Collapse into chunks
+    collapsed = []
+    for i in range(0, n, chunk_size):
+        chunk_df_bed_regions = df_bed_regions.iloc[i : i + chunk_size]
+        chrom = chunk_df_bed_regions.iloc[0]["chrom"]
+        start = chunk_df_bed_regions.iloc[0]["start"]
+        end = chunk_df_bed_regions.iloc[-1]["end"]
+        collapsed.append((chrom, start, end))
+
+    # Write output
+    genomic_intervals = []
+    for chrom, start, end in collapsed:
+        genomic_intervals.append(f"{chrom}:{start}-{end}")
+    return genomic_intervals
+
+
+def featuremap_fields_aggregation_on_an_interval_list(
+    featuremap_vcf_file: str,
+    output_vcf: str,
+    interval_list_bed_file: str,
+    filter_tags=None,
+    xgb_model_file: str = None,
+    verbose: bool = True,  # noqa: FBT001, FBT002
+) -> None:
+    """
+    Apply featuremap fields aggregation on an interval list
+    Inputs:
+        featuremap (str): The input featuremap vcf file
+        output_vcf (str): The output pileup vcf file
+        interval_list (str): The interval list file
```

Comment:
is it an interval_list or bed file? The parameter name suggests both in the function definition